inc/routing_table.h

```diff
@@ -93,4 +103,9 @@ uint16_t RoutingTB_GetLastService(void);
 uint16_t *RoutingTB_GetLastNode(void);
 uint16_t RoutingTB_GetLastEntry(void);
 
+// ********************* research routing table ********************************
+search_result_t *RTFilter_Type(search_result_t *result, luos_type_t type);
```

Comment:
If you put a `search_result_t *` as an argument, why do you return one too?
as I see it user should be able to do:

``` C
search_result_t result;
RTFilter_Type(&result, STATE);
```
And get result on the result var.

Reply:
we can do it like this, but we will need to call the functions serially and not like this: 
`RTFilter_Type(RTFilter_Node(0, 1), TYPE)`

Reply:
Yes. 
Perhaps we can do both by keeping your function like that but user need to pass the search_result_t (0 is not allowed)